spring-integration-core/src/main/java/org/springframework/integration/support/json/JacksonJsonUtils.java

```diff
@@ -43,38 +36,39 @@ private JacksonJsonUtils() {
 		super();
 	}
 
-	private static final ClassLoader classLoader = JacksonJsonUtils.class.getClassLoader();
-
-	private static final boolean jackson2Present =
-			ClassUtils.isPresent("com.fasterxml.jackson.databind.ObjectMapper", classLoader) &&
-					ClassUtils.isPresent("com.fasterxml.jackson.core.JsonGenerator", classLoader);
-
-	private static final boolean jacksonPresent =
-			ClassUtils.isPresent("org.codehaus.jackson.map.ObjectMapper", classLoader) &&
-					ClassUtils.isPresent("org.codehaus.jackson.JsonGenerator", classLoader);
-
+	/**
+	 * @return true if Jackson 2 is present in the classpath
+	 * @deprecated since 4.3.10 in favor of {@link JacksonPresent}
+	 */
+	@Deprecated
 	public static boolean isJackson2Present() {
-		return jackson2Present;
+		return JacksonPresent.isJackson2Present();
 	}
 
+	/**
+	 * @return true if Jackson 2 is present in the classpath
+	 * @deprecated since 4.3.10 in favor of {@link JacksonPresent}
+	 */
+
 	public static boolean isJacksonPresent() {
-		return jacksonPresent;
+		return JacksonPresent.isJacksonPresent();
 	}
 
 	/**
-	 * Return an {@link ObjectMapper} if available,
+	 * Return an {@link com.fasterxml.jackson.databind.ObjectMapper} if available,
 	 * supplied with Message specific serializers and deserializers.
 	 * Also configured to store typo info in the {@code @class} property.
 	 * @return the mapper.
 	 * @throws IllegalStateException if an implementation is not available.
 	 * @since 4.3.10
 	 */
-	public static ObjectMapper messagingAwareMapper() {
-		if (jackson2Present) {
-			ObjectMapper mapper = new ObjectMapper();
-			mapper.configure(MapperFeature.DEFAULT_VIEW_INCLUSION, false);
-			mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
-			mapper.enableDefaultTyping(ObjectMapper.DefaultTyping.NON_FINAL, JsonTypeInfo.As.PROPERTY);
+	public static com.fasterxml.jackson.databind.ObjectMapper messagingAwareMapper() {
+		if (JacksonPresent.isJackson2Present()) {
+			com.fasterxml.jackson.databind.ObjectMapper mapper = new com.fasterxml.jackson.databind.ObjectMapper();
```

Comment:
Doh... Do you do that on merge as well?
Thanks then.

During cherry-pick there will be some conflicts with the `ConfigurableCompositeMessageConverter` - it's a new `5.0` class.